### PR TITLE
fix(view): Fix closing nvim-tree float when file is removed

### LINKED ISSUE #1539: View float: Cannot close last window at remove open file
### Description

When deleting an open file with floating view enabled, I get an error.

### Neovim version

```text
NVIM v0.7.2
Build type: Release
LuaJIT 2.1.0-beta3
```

### Operating system and version

Linux arch 5.19.1-arch2-1

### nvim-tree version

e3353c4

### Minimal config

```lua
vim.cmd [[set runtimepath=$VIMRUNTIME]]
vim.cmd [[set packpath=/tmp/nvt-min/site]]
local package_root = "/tmp/nvt-min/site/pack"
local install_path = package_root .. "/packer/start/packer.nvim"
local function load_plugins()
  require("packer").startup {
    {
      "wbthomason/packer.nvim",
      "kyazdani42/nvim-tree.lua",
      "kyazdani42/nvim-web-devicons",
      -- ADD PLUGINS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
    },
    config = {
      package_root = package_root,
      compile_path = install_path .. "/plugin/packer_compiled.lua",
      display = { non_interactive = true },
    },
  }
end
if vim.fn.isdirectory(install_path) == 0 then
  print "Installing nvim-tree and dependencies."
  vim.fn.system { "git", "clone", "--depth=1", "https://github.com/wbthomason/packer.nvim", install_path }
end
load_plugins()
require("packer").sync()
vim.cmd [[autocmd User PackerComplete ++once echo "Ready!" | lua setup()]]
vim.opt.termguicolors = true
vim.opt.cursorline = true

-- MODIFY NVIM-TREE SETTINGS THAT ARE _NECESSARY_ FOR REPRODUCING THE ISSUE
_G.setup = function()
  local opts = { noremap = true, silent = true }
  vim.g.mapleader = " "
  vim.keymap.set("", "<Space>", "<Nop>", opts)
  vim.keymap.set("n", "<leader>e", ":NvimTreeRefresh<CR>:NvimTreeToggle<CR>", opts)
  
  require("nvim-tree").setup {
    sort_by = "case_sensitive",
    sync_root_with_cwd = true,
    respect_buf_cwd = true,
  
    update_focused_file = {
      enable = true,
      update_root = true
    },
    view = {
      float = {
        enable = true
      }
    }
  }
end
```

### Steps to reproduce

1. nvim -nu ~/nvt-min.lua
2. Open tree `<space>e`
3. Create file `a`
4. Open file
5. Open tree `<space>e`
6. Delete `d` current file

### Expected behavior

Deletes a file that is opened without an error

### Actual behavior

```
E5108: Error executing lua: ...t/nvim-tree.lua/lua/nvim-tree/actions/fs/remove-file.lua:12: Vim:E444: Cannot close last window
stack traceback:
        [C]: in function 'nvim_win_close'
        ...t/nvim-tree.lua/lua/nvim-tree/actions/fs/remove-file.lua:12: in function 'close_windows'
        ...t/nvim-tree.lua/lua/nvim-tree/actions/fs/remove-file.lua:29: in function 'clear_buffer'
        ...t/nvim-tree.lua/lua/nvim-tree/actions/fs/remove-file.lua:86: in function 'on_confirm'
        /usr/share/nvim/runtime/lua/vim/ui.lua:93: in function 'input'
        ...t/nvim-tree.lua/lua/nvim-tree/actions/fs/remove-file.lua:71: in function 'handle_tree_actions'
        ...r/start/nvim-tree.lua/lua/nvim-tree/actions/dispatch.lua:120: in function 'dispatch'
        ...acker/start/nvim-tree.lua/lua/nvim-tree/actions/init.lua:260: in function <...acker/start/nvim-tree.lua/lua/nvim-tree/actions/init.lua:259>
```

### PR DESCRIPTION
**Pull request by Khrees**
Resolves #1539

### CODE REVIEW

**Review by alex-courtis — APPROVED** [review]

File: lua/nvim-tree/actions/fs/remove-file.lua [hunk 1]
```diff
@@ -18,11 +23,13 @@ local function clear_buffer(absolute_path)
   local bufs = vim.fn.getbufinfo { bufloaded = 1, buflisted = 1 }
   for _, buf in pairs(bufs) do
     if buf.name == absolute_path then
-      if buf.hidden == 0 and #bufs > 1 then
+      if buf.hidden == 0 then
```
Comment (alex-courtis) [inline lua/nvim-tree/actions/fs/remove-file.lua:26]: What's the reason for this change?

It is a change in behaviour for non-floating:  the window will be closed even if there are other buffers present.

**Review by alex-courtis — CHANGES_REQUESTED** [review]
Love your work Khrees.

Tested:
* delete only open buffer
* delete current buffer
* delete other buffer
* delete unopened buffer

Last case leaves nvim-tree open which is reasonable behaviour.

- [x] https://github.com/kyazdani42/nvim-tree.lua/pull/1546/files#r956528621

File: lua/nvim-tree/actions/fs/remove-file.lua (hunk 1 above)
Comment (Khrees) [reply lua/nvim-tree/actions/fs/remove-file.lua:26]: I am not able to reproduce such behaviour. With non-floating window, it seems to behave the same regardless of this change. How can I reproduce this?

Comment (alex-courtis) [reply lua/nvim-tree/actions/fs/remove-file.lua:26]: > I am not able to reproduce such behaviour. With non-floating window, it seems to behave the same regardless of this change. How can I reproduce this?

I should have noted how to reproduce this as I now cannot... the next buffer is correctly displayed.

The window switch and `:bn` are now invoked when there are no other buffers in the window, however that seems fine.

**Real question:** do we need this change? If it's necessary for floating windows it should be done only for the case of floating windows.

Comment (Khrees) [reply lua/nvim-tree/actions/fs/remove-file.lua]: I reverted this change so non-floating windows should not be affected.

When I was trying to fix the original bug from the referenced issue, the deleted buffer was still displayed for some reason. The important bit here seems to be switching focus to window with soon-to-be-deleted buffer.

Comment (alex-courtis) [reply lua/nvim-tree/actions/fs/remove-file.lua]: > When I was trying to fix the original bug from the referenced issue, the deleted buffer was still displayed for some reason. The important bit here seems to be switching focus to window with soon-to-be-deleted buffer.

:+1:

**Pull request merged**